src/Toolkit/Toolkit.Maui/BasemapGallery/BasemapGallery.Appearance.cs

```diff
@@ -73,72 +98,91 @@ static BasemapGallery()
-            nameLabel.SetBinding(Label.TextProperty, nameof(BasemapGalleryItem.Name));
-            scrimGrid.SetBinding(OpacityProperty, nameof(BasemapGalleryItem.IsValid), mode: BindingMode.OneWay, converter: OpacityConverter);
-
-            border.Content = outerScrimContainer;
-            return border;
-#else
-            Border border = new Border { Padding = new Thickness(0), StrokeThickness = 1, StrokeShape = new Rectangle() };
-            Grid outerScrimContainer = new Grid();
-            outerScrimContainer.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(64) });
-            outerScrimContainer.ColumnDefinitions.Add(new ColumnDefinition { Width = new GridLength(8) });
-            outerScrimContainer.ColumnDefinitions.Add(new ColumnDefinition { Width = GridLength.Star });
-            outerScrimContainer.ColumnSpacing = 0;
-
-            Image thumbnail = new Image { WidthRequest = 64, HeightRequest = 64, Aspect = Aspect.AspectFill, HorizontalOptions = LayoutOptions.Start };
-            Label nameLabel = new Label { FontSize = 12, HorizontalTextAlignment = TextAlignment.Start, VerticalTextAlignment = TextAlignment.Center };
-
+#endif
+
             outerScrimContainer.Children.Add(thumbnail);
-            outerScrimContainer.Children.Add(nameLabel);
-
-            Grid.SetColumn(thumbnail, 0);
-            Grid.SetColumn(nameLabel, 2);
 
+            Grid.SetColumn(thumbnail, 0);
+
+            Grid newGrid = new Grid
+            {
+                Margin = new Thickness(0, 6, 6, 6)
+            };
+            newGrid.RowDefinitions.Add(new RowDefinition { Height = GridLength.Star });
+            newGrid.RowDefinitions.Add(new RowDefinition { Height = GridLength.Auto });
+
+            Label itemTextLabel = new Label
+            {
+                FontSize = 12,
+                HorizontalTextAlignment = TextAlignment.Start,
+                VerticalTextAlignment = TextAlignment.Center
+            };
+            itemTextLabel.SetBinding(Label.TextProperty, new Binding("Name", BindingMode.OneWay));
+            Grid.SetRow(itemTextLabel, 0);
+
+            Border itemTypeBorder = new Border
+            {
+                BackgroundColor = Colors.LightGray,
+                StrokeShape = new RoundRectangle { CornerRadius = new CornerRadius(7) },
+                Padding = new Thickness(4, 0),
+                HorizontalOptions = LayoutOptions.Start,
+                VerticalOptions = LayoutOptions.End,
+            };
+            itemTypeBorder.SetBinding(Border.IsVisibleProperty, new Binding("Is3D", BindingMode.OneWay));
+
+            Label itemTypeLabel = new Label
+            {
+                Text = "3D",
+                Margin = new Thickness(1),
+                HorizontalTextAlignment = TextAlignment.Center,
+                FontSize = 9,
+                TextColor = Colors.Black,

```

Comment:
Should this color adapt to darkmode?

Reply:
It must still adapt. We don't want black text in dark mode when everything else is white text.